wgengine/userspace_test.go

```diff
@@ -227,6 +231,86 @@ func TestUserspaceEnginePortReconfig(t *testing.T) {
 	}
 }
 
+// Test that enabling and disabling peer path MTU discovery works correctly.
+func TestUserspaceEnginePeerMTUReconfig(t *testing.T) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		t.Skipf("skipping on %q; peer MTU not supported", runtime.GOOS)
+	}
+
+	defer os.Setenv("TS_DEBUG_ENABLE_PMTUD", os.Getenv("TS_DEBUG_ENABLE_PMTUD"))
+	envknob.Setenv("TS_DEBUG_ENABLE_PMTUD", "")
+	// Turn on debugging to help diagnose problems.
+	defer os.Setenv("TS_DEBUG_PMTUD", os.Getenv("TS_DEBUG_PMTUD"))
+	envknob.Setenv("TS_DEBUG_PMTUD", "true")
+
+	var knobs controlknobs.Knobs
+
+	e, err := NewFakeUserspaceEngine(t.Logf, 0, &knobs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(e.Close)
+	ue := e.(*userspaceEngine)
+
+	if ue.magicConn.PeerMTUEnabled() != false {
+		t.Error("peer MTU enabled by default, should not be")
+	}
+	osDefaultDF, err := ue.magicConn.DontFragSetting()
+	if err != nil {
+		t.Errorf("get don't fragment bit failed: %v", err)
+	}
+	t.Logf("Info: OS default don't fragment bit(s) setting: %v", osDefaultDF)
+
+	// Build a set of configs to use as we change the peer MTU settings.
+	nodeKey, err := key.ParseNodePublicUntyped(mem.S("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg := &wgcfg.Config{
+		Peers: []wgcfg.Peer{
+			{
+				PublicKey: nodeKey,
+				AllowedIPs: []netip.Prefix{
+					netip.PrefixFrom(netaddr.IPv4(100, 100, 99, 1), 32),
+				},
+			},
+		},
+	}
+	routerCfg := &router.Config{}
+
+	tests := []struct {
+		wantP   bool     // desired value of PMTUD setting
+		wantDF  bool     // desired value of don't fragment bits
+		shouldP opt.Bool // if set, force peer MTU to this value
+		desc    string   // test description
+	}{
+		{wantP: false, wantDF: osDefaultDF, shouldP: "", desc: "after first reconfig"},
+		{wantP: true, wantDF: true, shouldP: "true", desc: "enabling PMTUD, first time"},
+		{wantP: false, wantDF: false, shouldP: "false", desc: "disabling PMTUD"},
+		{wantP: true, wantDF: true, shouldP: "true", desc: "enabling PMTUD, second time"},
+		{wantP: false, wantDF: false, shouldP: "", desc: "returning to default PMTUD"},
+	}
+
+	for _, tt := range tests {
+		t.Logf("--- testing %s", tt.desc)
```

Comment:
You might like `t.Run` here, though you'll want to swap spaces for underscores or similar in the test name, as go replaces space with underscore in the output

Reply:
Thanks so much for teaching me that! Fixed.